File: backend/api/schemas/investigation.py
from typing import Any

from pydantic import Field

from pydantic import BaseModel


class InvestigationStateSchema(BaseModel):
    investigation_id: int | None = None
    sensor_rows: list[dict[str, Any]] = Field(default_factory=list)
    risk_level: str = ""
    anomalies: list[Any] = Field(default_factory=list)
    incidents: list[Any] = Field(default_factory=list)
    github_matches: list[Any] = Field(default_factory=list)
    root_causes: list[Any] = Field(default_factory=list)
    recommendations: list[Any] = Field(default_factory=list)
    summary: str = ""
    summary_text: str = ""
    summary_sections: dict[str, Any] = Field(default_factory=dict)
    github_match_status: str = ""
    github_issue_status: str = ""
    github_issue_id: int | None = None
    github_issue_url: str = ""
    github_issue_detail: str = ""
    technical_report_status: str = ""
    technical_report_filename: str = ""
    technical_report_path: str = ""
    technical_report_preview: str = ""
    failure_summary: str = ""


class InvestigationRunResponse(BaseModel):
    investigation_id: int | None = None
    upload_id: str
    status: str
    state: InvestigationStateSchema

File: frontend/src/types/investigation.ts
export interface InvestigationState {
  risk_level?: string;
  anomalies: unknown[];
  incidents: unknown[];
  github_matches?: unknown[];
  github_match_status?: string;
  root_causes: unknown[];
  recommendations: unknown[];
  summary: string;
  summary_text?: string;
  summary_sections?: StructuredSummary;
  github_issue_status?: string;
  github_issue_id?: number | null;
  github_issue_url?: string;
  github_issue_detail?: string;
  investigation_id?: number | null;
  technical_report_status?: string;
  technical_report_filename?: string;
  technical_report_path?: string;
  technical_report_preview?: string;
}

export interface InvestigationRun {
  investigation_id: number;
  upload_id: string;
  status: string;
  state: InvestigationState;
}

export interface StructuredSummary {
  headline?: string;
  overview?: string;
  anomalies?: string[];
  historical_matches?: Array<{
    incident_id?: number;
    similarity?: number;
    failure?: string;
    root_cause?: string;
    issue_url?: string;
  }>;
  root_causes?: Array<{
    cause?: string;
    confidence?: number;
    evidence?: string[];
  }>;
  recommendations?: string[];
  action_plan?: string[];
  summary_text?: string;
}
